src/screens/SettingsModal.js

```diff
@@ -54,17 +53,35 @@ const requestFaceIDPermission = () =>
       captureException(error);
     });
 
-const SettingsModal = ({
-  currentSettingsPage,
-  navigation,
-  onCloseModal,
-  onPressBack,
-  onPressImportSeedPhrase,
-  onPressSection,
-}) => {
+const SettingsModal = ({ navigation }) => {
+  const currentSettingsPage = get(
+    navigation,
+    'state.params.section',
+    SettingsPages.default
+  );
+
   const { component, title } = currentSettingsPage;
   const isDefaultPage = title === SettingsPages.default.title;
 
+  const onCloseModal = useCallback(() => navigation.goBack(), [navigation]);
+
+  const onPressBack = useCallback(
+    () => navigation.setParams({ section: SettingsPages.default }),
+    [navigation]
+  );
+
+  const onPressImportSeedPhrase = useCallback(() => {
+    InteractionManager.runAfterInteractions(() => {
+      navigation.navigate(Routes.IMPORT_SEED_PHRASE_SHEET);
+      StatusBar.setBarStyle('light-content');
```

Comment:
I'd keep all the statusbar logic in the same place (Routes.ios | android ) => onNavigationStateChange

Reply:
Will do this separately!